dali/kernels/audio/mel_scale/mel_filter_bank_gpu.cu

```diff
@@ -0,0 +1,311 @@
+                                     norm_factors, fftbin_start_, fftbin_end_, args_.nfilter);
+    }
+  }
+
+  using MelFilterImplBase<T, Dims>::Args;
+
+ private:
+  void SetupBlockDescsOuterFft(const TensorListShape<Dims> &in_shape, ScratchpadEstimator &se) {
+    auto batch_size = in_shape.num_samples();
+    for (int64_t ti = 0; ti < batch_size; ++ti) {
+      const auto &tshape = in_shape.tensor_shape(ti);
+      if (args_.axis > 0)
+        nframes_.push_back(volume(tshape.data(), tshape.data() + args_.axis));
+      else
+        nframes_.push_back(1);
+      nwindows_.push_back(volume(tshape.data() + args_.axis + 1, tshape.data() + tshape.size()));
+      for (int64_t s = 0; s < nframes_.back(); ++s) {
+        auto nblocks = (nwindows_.back() + kBlockDim2 - 1) / kBlockDim2;
+        for (int64_t b = 0; b < nblocks; ++b) {
+          block_descs_.push_back(BlockDesc<T>{nullptr, nullptr,
+                                              {b * kBlockDim2, nwindows_.back()}});
+        }
+      }
+    }
+    se.add<BlockDesc<T>>(AllocType::GPU, block_descs_.size());
+  }
+
+  void SetupBlockDescsInnerFft(const TensorListShape<Dims> &in_shape, ScratchpadEstimator &se) {
+    auto batch_size = in_shape.num_samples();
+    for (int64_t ti = 0; ti < batch_size; ++ti) {
+      const auto &tshape = in_shape.tensor_shape(ti);
+      auto sample_size = volume(tshape.begin(), tshape.end() - 1) * args_.nfilter;
+      auto nblocks = (sample_size + kBlockDim1 - 1) / kBlockDim1;
+      for (int b = 0; b < nblocks; ++b) {
+        block_descs_.push_back(BlockDesc<T>{nullptr, nullptr, {b * kBlockDim1, sample_size}});
+      }
+    }
+    se.add<BlockDesc<T>>(AllocType::GPU, block_descs_.size());
+  }
+
+  void FillBlockDescsOuterFft(const T* const* in_list, T **out_list) {
+    int64_t block_id = 0;
+    for (uint64_t ti = 0; ti < nframes_.size(); ++ti) {
+      const T *in = in_list[ti];
+      T *out = out_list[ti];
+      for (int64_t s = 0; s < nframes_[ti]; ++s) {
+        auto nblocks = (nwindows_[ti] + kBlockDim2 - 1) / kBlockDim2;
+        for (int b = 0; b < nblocks; ++b) {
+          block_descs_[block_id].in_frame = in;
+          block_descs_[block_id].out_frame = out;
+          ++block_id;
+        }
+        in += nwindows_[ti] * fft_dim_;
+        out += nwindows_[ti] * args_.nfilter;
+      }
+    }
+  }
+
+  void FillBlockDescsInnerFft(const T* const* in_list, T **out_list) {
+    int block_id = 0;
+    int sample = 0;
+    while (block_id < block_descs_.size()) {
+      auto sample_size = block_descs_[block_id].out_frame_size;
+      auto nblocks = (sample_size + kBlockDim1 - 1) / kBlockDim1;
+      for (int i = 0; i < nblocks; ++i, ++block_id) {
+        block_descs_[block_id].in_frame = in_list[sample];
+        block_descs_[block_id].out_frame = out_list[sample];
+      }
+      ++sample;
+    }
+  }
+
+  std::vector<int64_t> interval_ends_;
```

Comment:
It's next to impossible that intervals in FFT bin domain would need 64-bits - and for GPU 64-bit integers are slower, AFAIK.

Reply:
done